### PR TITLE
feat(network_info_plus): Add Swift Package Manager support

### PR DESCRIPTION
**Pull request by charafau**
ref: https://github.com/fluttercommunity/plus_plugins/issues/3152

Another spm support

### CODE REVIEW

**Review by miquelbeltran — COMMENTED** [review]
LGTM, still want to test on my device before approval and merge

**Review by vbuberen — APPROVED** [review]
Fixed the privacy manifest location and declaration for MacOS. 
Unfortunately, I currently don't have a personal account with Wi-Fi capability, so couldn't test this one, but considering that all other PRs on this topic are alright, I believe we are good to go.

**Pull request merged**
